tasks/linux.sh

```diff
@@ -161,4 +113,37 @@ case "$available_manager" in
         cmd_status="$(yum_status "\{ \"old_version\": ${old_version}, \"version\": \"%{VERSION}-%{RELEASE}\" \}")"
```

Comment:
Need `old_version` for sles.

Reply:
From what I can tell the refactor did not break any existing tests for linux targets. You will need to update the sles `update` to contain the key `old_version` for the sles targets to pass.

Reply:
Added.  I had to rewrite a chunk of the `zypper` functionality because of a quirk of how it updates packages.  If the newer version comes from a different vendor, `zypper up` won't actually upgrade the package.  You have to do `zypper in <package>-<version>`, so I went ahead and did that.  Otherwise `status` would tell you a newer version is available, but `upgrade` wouldn't actually upgrade it.

Reply:
sounds good. I'm assuming the acceptance tests pass given we only validate the expected keys are there. Could in also test that the result of those keys are not empty strings? (thinking of the amazon linux service task example here)

Reply:
I don't think we'll have that issue we're checking for returns codes for `rpm` and `dpkg`, and checking for an empty string for `zypper`.  A quick test for uninstalled packages on each gives `"status": "uninstalled"`